gym/spaces/utils.py

```diff
@@ -43,9 +43,9 @@ def flatten(space, x):
     ``gym.spaces``.
     """
     if isinstance(space, Box):
-        return np.asarray(x, dtype=np.float32).flatten()
+        return np.asarray(x, dtype=space.dtype).flatten()
```

Comment:
using space.dtype  here instead of np.int64 seems inconsistent with the clause for MultiDiscrete and MultiBinary

Reply:
I originally did this because the underlying dtype of Box can be defined by the user, but the types of discrete, multi_discrete and multi_binary are hardcoded to certain values.

Reply:
I will change them all to be space.dtype for consistency. this will also accommodate for if anyone chooses to change the underlying types of multi_discrete, discrete,  multi_binary (which i found has actually been done a couple times before).